experiments/202006-redux-lithtml-codemirror/index.html

```diff
@@ -0,0 +1,215 @@
+    But what is this "logic" doing?
+    In Redux, it is simply transforming the Model in some way
+
+      function controller(currentModel, action) {
+        switch (action.type) {
+
+          case "rename":
+            return {...currentModel, name: action.newName}
+
+          case "clear":
+            return {...currentModel, content: ""}
+
+          ...
+        }
+      }
+
+
+    You hand this method to Redux, and internally it's taking actions that happen (say as a result
+    of something a user does with your app), and calling this big switch method...
+
+      const newModel = controller(model, {type:"rename", newName: "foo2"})
+
+    Current model + action go in, new model comes out. Repeat.
+
+
+
+    Nomenclature change: Redux has different terms though. Model = State, Controller = Reducer. Revised example,
+      in which your "function controller(..." is now "function myReducer(...":
+
+      const newState = myReducer(state, {type:"rename", newName: "foo2"})
+
+    Immutability: it's pretty important that the incoming state object not be side-affected, at least if you want to
+    keep your sanity when debugging stuff.  You can simply use object spread (as above). There's also immer,
+    which I like and which the redux folks appear to still recommend https://stackoverflow.com/a/60819732
+
+
+
+    The "V" in MVC.
+
+    Redux has something it calls a store, you create it using your reducer function and an initial state:
+
+    const initialState = {name: "foo"}
+    const store = Redux.createStore(myReducer, initialState)  // i.e. "controller" and initial "model"
+
+      // think: simplest possible "view rendering function"...but just for debugging purposes
+      function nameConsoleRenderer() {
+        console.log(`current name: ${store.getState().name}`)
+      }
+
+      // think: simplest possible html-based "view rendering function"
+      function nameDomRenderer() {
+        document.body.innerHTML = `Name: <br>${store.getState().name}</br>`
+      }
+
+    Then you add your rendering functions to the store:
+
+      store.subscribe(nameConsoleRenderer)
+      store.subscribe(nameDomRenderer)
+
+    ...and every time there's a state update, redux will invoke each view function.
+    Your immediate reaction is likely:
+      "holy crap does that mean html-re-rendering is happening all the time? That's nuts."
+    To which I say:
+      - "No - whether to actually have dom-mutating effects on any invocation is up to the view function.
+        You can have it hold its own internal state across view invocations (e.g. via a higher-order function that closes
+        around some common tracking variables), which you will often do. For example, it's totally unacceptable to
+        re-render a code editor as someone is typing in it."
+      ...but also
+      - "Yes - by default I let things just re-render...and that's because of stuff like lit-html, that uses
```

Comment:
I'm pretty sure there is a way in the redux store to subscribe to a specific area in an object graph, did a little googling.